wrappers/src/results_cs.cpp

```diff
@@ -124,20 +123,19 @@ REALM_EXPORT Results* results_get_filtered_results(const Results& results, uint1
 {
     return handle_errors(ex, [&]() {
         Utf16StringAccessor query_string(query_buf, query_len);
-        auto query = results.get_query();
-        auto const &realm = results.get_realm();
+        auto const& realm = results.get_realm();
 
-        parser::ParserResult result = parser::parse(query_string.to_string());
-
-        parser::KeyPathMapping mapping;
+        query_parser::KeyPathMapping mapping;
         realm::populate_keypath_mapping(mapping, *realm);
 
-        query_builder::NoArguments no_args;
-        query_builder::apply_predicate(query, result.predicate, no_args, mapping);
+        query_parser::NoArguments no_args;
+        Query parsed_query = results.get_table()->query(query_string, no_args, mapping);
+        DescriptorOrdering new_order = results.get_descriptor_ordering();
+        if (auto parsed_ordering = parsed_query.get_ordering()) {
+            new_order.append(*parsed_ordering);
+        }
 
-        DescriptorOrdering ordering;
-        query_builder::apply_ordering(ordering, query.get_table(), result.ordering);
-        return new Results(realm, std::move(query), std::move(ordering));
+        return new Results(realm, results.get_query().and_query(std::move(parsed_query)), new_order);
```

Comment:
`std::move(new_order)`?

Reply:
I don't know, to be honest. I noticed that as well but @irongage didn't use it in the example. I thought it was a stylistic error, but then I went to check and only `Query` has a custom move constructor, while `DescriptorOrdering` has only the default one. After seeing that I thought that it was not a mistake but a wanted choice.
What is your opinion?

Reply:
I don't know - I'm not a C++ dev 😄 As mentioned, I always add `std::move` when I want to move something, just to be on the safe side. Maybe @ironage can educate us on the pros/cons of `std::move`-ing the ordering?

Reply:
That feedback would be very welcome. 🙂

Reply:
Yes you can move the ordering, I just forgot. Either way is functionally correct, but moving it is an efficient transfer of ownership, while not moving it invokes the [copy constructor ](https://github.com/realm/realm-core/blob/master/src/realm/sort_descriptor.cpp#L545-L550) which inefficiently makes copies of all the contents into the new object only to then discard the previous object entirely.

Reply:
My only doubt was coming from the fact that while brushing on `std::move` I read that returning automatic local variables generally (not sure if always) uses the move semantics instead of the copy ones, as long as a move constructor and assignment exist. So I was not sure if stylistically people still added a `std::move` on the returned values. You clearly answered that people generally prefer to be explicit, thank you for the feedback.